pkg/registry/generic/etcd/etcd_test.go

```diff
@@ -227,29 +176,30 @@ func TestEtcdList(t *testing.T) {
 		if item.context != nil {
 			ctx = item.context
 		}
-		fakeClient, registry := NewTestGenericEtcdRegistry(t)
-		if name, ok := item.m.MatchesSingle(); ok {
-			if key, err := registry.KeyFunc(ctx, name); err == nil {
-				key = etcdtest.AddPrefix(key)
-				fakeClient.Data[key] = item.in
-			} else {
-				key := registry.KeyRootFunc(ctx)
-				key = etcdtest.AddPrefix(key)
-				fakeClient.Data[key] = item.in
+		server, registry := NewTestGenericEtcdRegistry(t)
+
+		if item.in != nil {
+			if err := etcdstorage.CreatePodList(t, "/pods", registry.Storage, item.in); err != nil {
+				t.Errorf("Unexpected error %v", err)
 			}
-		} else {
-			key := registry.KeyRootFunc(ctx)
-			key = etcdtest.AddPrefix(key)
-			fakeClient.Data[key] = item.in
 		}
+
 		list, err := registry.ListPredicate(ctx, item.m, nil)
-		if e, a := item.succeed, err == nil; e != a {
-			t.Errorf("%v: expected %v, got %v: %v", name, e, a, err)
+		if err != nil {
+			t.Errorf("Unexpected error %v", err)
 			continue
 		}
-		if e, a := item.out, list; !api.Semantic.DeepDerivative(e, a) {
-			t.Errorf("%v: Expected %#v, got %#v", name, e, a)
+		if item.out != nil {
+			if e, a := item.out, list; !api.Semantic.DeepDerivative(e, a) {
```

Comment:
Using DeepDerivative is pretty tricky - in fact I found a bug with that ~week ago - see: #16020

Can you please add an explicit comment that we need to check DeepDerivative(e,a) not (a,e) and that we use DeepDerivative because storage layer sets resource versions?